crates/astria-sequencer/src/service/mempool.rs

```diff
@@ -25,12 +26,14 @@ use tendermint::v0_38::abci::{
     MempoolRequest,
     MempoolResponse,
 };
+use tokio::sync::Mutex;
```

Comment:
Do you expect to hold the Mutex across await points? If not, it is recommended to use `std::sync::Mutex`

Reply:
unfortunately i do, in the `update_mempool_after_finalization` method.